__tests__/__unit__/uss/ZoweUSSNode.unit.test.ts

```diff
@@ -0,0 +1,617 @@
+            "testDir", vscode.TreeItemCollapsibleState.Collapsed, rootNode, null, null, false, profileOne.name, undefined);
+        const testFile = new ZoweUSSNode(
+            "testFile", vscode.TreeItemCollapsibleState.None, testDir, null, null, false, profileOne.name, undefined);
+        testFile.contextValue = DS_TEXT_FILE_CONTEXT;
+        expect(JSON.stringify(rootNode.iconPath)).toContain("folder-closed.svg");
+        expect(JSON.stringify(testDir.iconPath)).toContain("folder-closed.svg");
+        expect(JSON.stringify(testFile.iconPath)).toContain("document.svg");
+        rootNode.iconPath = "Ref: 'folder.svg'";
+        testDir.iconPath = "Ref: 'folder.svg'";
+        testFile.iconPath = "Ref: 'document.svg'";
+        expect(testFile).toMatchSnapshot();
+    });
+
+    it("Tests that creating a new USS node initializes all methods and properties", async () => {
+        const testNode = new ZoweUSSNode(
+            "/u", vscode.TreeItemCollapsibleState.None, null, session, null, false, profileOne.name, undefined);
+        testNode.contextValue = USS_SESSION_CONTEXT;
+
+        expect(testNode.label).toBeDefined();
+        expect(testNode.collapsibleState).toBeDefined();
+        expect(testNode.label).toBeDefined();
+        expect(testNode.getParent()).toBeDefined();
+        expect(testNode.getSession()).toBeDefined();
+    });
+});
+
+describe("ZoweUSSNode Unit Tests - Function node.getSession()", () => {
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it("Tests that node.getSession() returns the proper session", async () => {
+        // Creating a rootNode
+        const rootNode = new ZoweUSSNode(
+            "root", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
+        rootNode.contextValue = USS_SESSION_CONTEXT;
+        const subNode = new ZoweUSSNode(
+            DS_PDS_CONTEXT, vscode.TreeItemCollapsibleState.Collapsed, rootNode, null, null, false, profileOne.name, undefined);
+        const child = new ZoweUSSNode(
+            "child", vscode.TreeItemCollapsibleState.None, subNode, null, null, false, profileOne.name, undefined);
+
+        const returnedSession = child.getSession();
+        expect(returnedSession).toBeDefined();
+        expect(returnedSession).toStrictEqual(session);
+    });
+});
+
+describe("ZoweUSSNode Unit Tests - Function node.refreshUSS()", () => {
+    const ussNode = new ZoweUSSNode("usstest", vscode.TreeItemCollapsibleState.Expanded, null, session, null, null, profileOne.name, "123");
+    ussNode.contextValue = USS_SESSION_CONTEXT;
+    ussNode.fullPath = "/u/myuser";
+    let node;
+
+    // USS favorited node definition
+    const ussNodeFav = new ZoweUSSNode("[profile]: usstest", vscode.TreeItemCollapsibleState.Expanded, null, session, null, false, profileOne.name);
+    ussNodeFav.contextValue = DS_TEXT_FILE_CONTEXT + FAV_SUFFIX;
+    ussNodeFav.fullPath = "/u/myuser/usstest";
```

Comment:
Do you think it is wise to move the hard-coded values to globals as well?

Reply:
I am concerned that, if these variables are available to all blocks, one block might change their properties, and then all subsequent blocks would be affected. For example `contextValue `or `fullPath `could be set anywhere.

So to prevent this, my options were to declare it in the describe block, or to declare it with globals and reset it with` afterAll() `in every describe block. Since this variable is only used in this` describe() `block it made more sense to put it here than to require every block to reset the variable.

Maybe you know a better way to do it? I'll try other options for sure.

Reply:
no I have no other way :) Leave it as is if you think that is the best.